Rubberduck.Inspections/Concrete/UnreachableCaseInspection.cs

```diff
@@ -0,0 +1,1309 @@
+
+                if ( !selectStmtDO.CanBeInspected) { continue; }
+
+                selectStmtDO = InitializeCaseClauses(selectStmtDO);
+
+                if (!selectStmtDO.CanBeInspected) { continue; }
+
+                selectStmtDO = InspectSelectStmtCaseClauses(selectStmtDO);
+
+                if (!selectStmtDO.CanBeInspected) { continue; }
+
+                inspResults.AddRange(selectStmtDO.CaseClauseDOs.Where(cc => cc.ResultType != ClauseEvaluationResult.NoResult)
+                    .Select(cc => CreateInspectionResult(selectStmt, cc.CaseContext, _resultMessages[cc.ResultType])));
+
+                if (selectStmtDO.SummaryCaseClauses.CaseElseIsUnreachable && selectStmtDO.CaseElseContext != null)
+                {
+                    inspResults.Add(CreateInspectionResult(selectStmt, selectStmtDO.CaseElseContext, _resultMessages[ClauseEvaluationResult.CaseElse]));
+                }
+            }
+            return inspResults;
+        }
+
+        private SelectStmtDataObject InitializeSelectStatementDataObject(SelectStmtDataObject selectStmtDO)
+        {
+            selectStmtDO = ResolveSelectStmtInspectionType(selectStmtDO);
+
+            if (selectStmtDO.CanBeInspected)
+            {
+                selectStmtDO.CaseClauseDOs = ParserRuleContextHelper.GetChildren<CaseClauseContext>(selectStmtDO.SelectStmtContext)
+                    .Select(cc => CreateCaseClauseDataObject(cc, selectStmtDO.BaseTypeName)).ToList();
+            }
+            return selectStmtDO;
+        }
+
+        private SelectStmtDataObject ResolveSelectStmtInspectionType(SelectStmtDataObject selectStmtDO)
+        {
+            if (!ContextCanBeEvaluated(selectStmtDO.SelectExpressionContext, selectStmtDO.IdReferenceName))
+            {
+                return InferTheSelectStmtType(selectStmtDO);
+            }
+
+            if (ParserRuleContextHelper.GetDescendents(selectStmtDO.SelectExpressionContext)
+                .Any(desc => IsBinaryLogicalOperation(desc) || IsUnaryLogicalOperator(desc)))
+            {
+                return SetTheTypeNames(selectStmtDO, Tokens.Boolean);
+            }
+
+            var firstLExpr = ParserRuleContextHelper.GetDescendent<LExprContext>(selectStmtDO.SelectExpressionContext);
+            if(firstLExpr == null)
+            {
+                return InferTheSelectStmtType(selectStmtDO);
+            }
+
+            var expression = ParserRuleContextHelper.GetDescendent<SimpleNameExprContext>(firstLExpr).GetText();
+            if (SymbolList.TypeHintToTypeName.ContainsKey(expression.Last().ToString()))
+            {
+                return SetTheTypeNames(selectStmtDO, SymbolList.TypeHintToTypeName[expression.Last().ToString()]);
+            }
+
+            var idRefs = (State.DeclarationFinder.MatchName(expression).Select(dec => dec.References))
+                .SelectMany(rf => rf).Where(idr => ParserRuleContextHelper.HasParent(idr.Context, selectStmtDO.SelectExpressionContext));
```

Comment:
ParserRuleContextHelper seems like it should be extension methods... `idr.Context.HasParent(selectStmtDO.SelectExpressionContext)` would be a little cleaner

Reply:
Pretty sure that is an extension method already. If it isn't, Vogel's right.

Reply:
Added extension method(s)